src/clp_ffi_py/ir/native/PyMetadata.cpp

```diff
@@ -237,13 +239,8 @@ auto PyMetadata::init(nlohmann::json const& metadata, bool is_four_byte_encoding
     try {
         // NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
         m_metadata = new Metadata(metadata, is_four_byte_encoding);
-    } catch (ExceptionFFI const& ex) {
-        PyErr_Format(
-                PyExc_RuntimeError,
-                "Failed to initialize metadata from deserialized JSON format preamble. "
-                "Error message: %s",
-                ex.what()
-        );
+    } catch (clp::TraceableException& ex) {
```

Comment:
One last comment:

By replacing ExceptionFFI with clp::TraceableException, now we need to use dynamic cast in the handle_traceable_exception.

Just by looking at this PR, I don't see the benefit of replacing ExceptionFFI with a more general clp::TraceableException, but I guess you have some insights/future plan?

Reply:
hmmm, for this one I think it is fine to keep using `clp_ffi_py::ExceptionFFI`, and maybe we should since it's the only type of exception it could throw. In future PRs, we may also need to capture exceptions from clp core code execution, so it's better to catch a generic type instead.
Changed it back to `clp_ffi_py::ExceptionFFI` in the latest push